### PR TITLE
Small QoL improvement on smartfridge item release

### PR DESCRIPTION
**Pull request by yorii**
[Changelogs]: 
:cl: yorii
fix: Fixed the smartfridges to be in line with all other machines and puts the released item in your hand.
/:cl:

### CODE REVIEW

**Review by SpaceManiac — DISMISSED** [review]

**Comment by DaxDupont** [comment]
oh yeah i forgot to do this

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
If you pick Many and enter a number bigger than how many there actually are, none of them get put in your hands. It might be reasonable to simply attempt putting the item in hand every time - so vending 2 with two empty hands picks up both of them.

**Comment by ExcessiveUseOfCobblestone** [comment]
In what instance would you drop say 5 of the same item but want 1/2 of those in your hands? You'd want all of them in a plant bag or something similar. If you choose more than 2 it should all go to the ground (you can get by with this by clicking 1,optional 1,many which are right beside each other)

It wouldn't exactly be a QoL if you had to take more clicks than what you did previously for a common event (you want items to go to a plant bag).

**Comment by yorii** [comment]
Hmm, differing opinions here, how about I only make the withdraw of a single item put it in your hand and anything else is ejected on the tile?

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 1]
```diff
@@ -185,7 +185,10 @@
 					break
 				if(O.name == params["name"])
 					O.forceMove(drop_location())
-					adjust_item_drop_location(O)
+					if(desired == 1 && Adjacent(usr) && !issilicon(usr))
```
Comment (yorii) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: If I just edit this line to `if(Adjacent(usr) && !issilicon(usr))` will the `usr.put_in_hands(O)` drop the items on the ground if no hands are free?

**Comment by ExcessiveUseOfCobblestone** [comment]
Perhaps you can make it put in hands if Max or Equal to your amount of arms rather than a specific 2 then. I see what he is saying and I agree it should go into your hands UNLESS you want way more items than you could realistically carry, which you'd want on the ground for single click bags which is applicable to several variants of the fridge.

**Comment by SpaceManiac** [comment]
`put_in_hands` with an S will place the item on the floor beneath the mob if it fails. You can check if it returns `FALSE` and put it back on the same tile as the fridge with the usual pixel placement logic.

I think making "One" try to put in hands and "Many" always drop on tile would be best. If you want two in your hands you can click "One" twice.

**Comment by yorii** [comment]
>I think making "One" try to put in hands and "Many" always drop on tile would be best. If you want two in your hands you can click "One" twice.

Yeah okay, I'll do this.

**Pull request merged**